game-app/game-core/src/test/java/games/strategy/ui/UtilTest.java

```diff
@@ -2,12 +2,37 @@
 
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 import java.awt.Polygon;
+import javax.swing.SwingUtilities;
 import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
 
 final class UtilTest {
+  @Test
+  void ensureOnEventDispatchThread() {
+    new Thread(
+            () ->
+                assertAll(
+                    () -> SwingUtilities.invokeAndWait(() -> Util.ensureOnEventDispatchThread())))
+        .start();
```

Comment:
Wrapping assert statements inside a new thread will cause the test to pass even if the assertion fails.
Just remove the new Thread() part around it and it should work as expected. Same for the other method